src/repository/rest/rickandmortyapi.go

```diff
@@ -0,0 +1,88 @@
+package rest
+
+import (
+	"encoding/json"
+	"github.com/go-resty/resty/v2"
+	"github.com/spf13/viper"
+	"golang-bootcamp-2020/domain/model"
+	_errors "golang-bootcamp-2020/utils/error"
+)
+
+var (
+	apiCharacters = "https://rickandmortyapi.com/api/character/"
+)
+
+type restResponse struct {
+	Info    paginationInfo `json:"info"`
+	Results []interface{}  `json:"results"`
+}
+
+type paginationInfo struct {
+	Next string `json:"next"`
+}
+
+type rickAndMortyApi struct {
+}
+
+type RickAndMortyApiRepository interface {
+	FetchData(maxPages int) ([]model.Character, _errors.RestError)
+}
+
+//Return new rest repository
+func NewRickAndMortyApiRepository() RickAndMortyApiRepository {
+	return &rickAndMortyApi{}
+}
+
+//Fetch data from rick and morty api
+func (api *rickAndMortyApi) FetchData(maxPages int) ([]model.Character, _errors.RestError) {
+	// fetching characters
+	var characters []model.Character
+	resp, err := processRequest(apiCharacters, maxPages)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range resp {
+		var ch []model.Character
+		json.Unmarshal(resp[i], &ch)
+		characters = append(characters, ch...)
+	}
+
+	return characters, nil
+}
+
+func processRequest(url string, maxPages int) ([][]byte, _errors.RestError) {
+
+	var response [][]byte
+	endpoint := url
+	count := 1
+
+	if maxPages == 0 {
+		maxPages = viper.GetInt("rest.maxPagesByDefault")
+	}
+
+	for {
+		client := resty.New()
```

Comment:
The services or clients should be created in the main file, and pass the service to the layer will use it
This will help, for example:

- In the main, you create the services, and check for the service's healthiness, if something is wrong, the application shouldn't start.

- If you need to add more services like this one in the future, you only need to send another host to the `New` function and you will create another service, without repeating code or changing anything in another layer.